framework/src/modules/chain/transport/transport.js

```diff
@@ -254,8 +291,11 @@ class Transport {
 		const errors = validator.validate(schemas.signaturesList, query);
 
 		if (errors.length) {
-			this.logger.debug({ err: errors }, 'Invalid signatures body');
-			// TODO: If there is an error, invoke the applyPenalty action on the Network module once it is implemented.
+			this.logger.warn({ err: errors }, 'Invalid signatures body');
```

Comment:
Why don't instead of calling applyPenalty when an invalid input is received, we throw a specific error that is caught in the upper scope and then we apply penalty there? I see this is a repeating pattern and we could avoid code repetition.

Reply:
Actually, this cannot be done since each function is exposed as a action, and there is no outer function than controller now